Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp

```diff
@@ -44,13 +48,73 @@
 #include <ostream>
 #include <stdexcept>
 #include <system_error>
+#include <unordered_set>
 #include <utility>
 
-#include <boost/histogram.hpp>
+#include <boost/functional/hash.hpp>
 
-ActsExamples::TrackFindingAlgorithm::TrackFindingAlgorithm(
-    Config config, Acts::Logging::Level level)
-    : ActsExamples::IAlgorithm("TrackFindingAlgorithm", level),
+namespace ActsExamples {
+namespace {
+
+/// Source link indices of the bottom, middle, top measurements.
+/// In case of strip seeds only the first source link of the pair is used.
+using SeedIdentifier = std::array<Index, 3>;
+
+SeedIdentifier makeSeedIdentifier(const SimSeed& seed) {
+  SeedIdentifier result;
+
+  for (const auto& [i, sp] : Acts::enumerate(seed.sp())) {
+    const Acts::SourceLink& firstSourceLink = sp->sourceLinks().front();
+    result.at(i) = firstSourceLink.get<IndexSourceLink>().index();
+  }
+
+  return result;
+}
+
+template <typename Visitor>
+void visitSeedIdentifiers(const TrackProxy& track, Visitor visitor) {
+  std::vector<Index> sourceLinkIndices;
+  sourceLinkIndices.reserve(track.nMeasurements());
+  for (const auto& trackState : track.trackStatesReversed()) {
+    if (!trackState.hasUncalibratedSourceLink()) {
+      continue;
+    }
+    const Acts::SourceLink& sourceLink = trackState.getUncalibratedSourceLink();
+    sourceLinkIndices.push_back(sourceLink.get<IndexSourceLink>().index());
+  }
+
+  for (std::size_t i = 0; i < sourceLinkIndices.size(); ++i) {
+    for (std::size_t j = i + 1; j < sourceLinkIndices.size(); ++j) {
+      for (std::size_t k = j + 1; k < sourceLinkIndices.size(); ++k) {
+        // Putting them into reverse order (k, j, i) to compensate for the
+        // `trackStatesReversed` above.
+        visitor({sourceLinkIndices.at(k), sourceLinkIndices.at(j),
+                 sourceLinkIndices.at(i)});
+      }
+    }
+  }
+}
```

Comment:
Correct me if I am wrong but here what you plan to do is to create all the possible seed associated to a track then check if a new seed belong to that list. Is that really much faster than checking if the 3 hits belong to the list of measurements ?

Reply:
And If I am correct it might be good to add a comment explaining the logic here

Reply:
This is definitely not cheap in terms of memory and CPU but we thought it would be better than checking either every seed against all reconstructed tracks or every track against all seeds.

The difference being that we build up a lookup table for seeds after each track without actually looping through tracks or seeds. So this should scale better compared to checking all seeds or tracks.

Reply:
At first I was thinking that you could each time you are adding a track compare the remaning seed with it and remove the matching ones. That way you don't need to store anything. But that would mean modifying the seed list while iterating over it which is also not nice.

Reply:
In any case, I think this logic need to be explain in a comment somewhere since it is not fully obvious what the code does

Reply:
> At first I was thinking that you could each time you are adding a track compare the remaning seed with it and remove the matching ones. That way you don't need to store anything. But that would mean modifying the seed list while iterating over it which is also not nice.

This is basically what is happening but without introducing another loop on the tracks/seeds but rather creating a lookup table for discovered triplets

> In any case, I think this logic need to be explain in a comment somewhere since it is not fully obvious what the code does

Makes sense - I will add some doc